### PR TITLE
Enable all cloud-init services on boot (all OS)

### PR DESCRIPTION
**Pull request by EleanorRigby**
Signed-off-by: Tushar Aggarwal <taggarwal@vmware.com>

/assign @codenrhoden 

fixes partly #129

### CODE REVIEW

File: images/capi/ansible/roles/providers/tasks/main.yml [hunk 1]
```diff
@@ -27,3 +27,21 @@
 
 - include_tasks: qemu.yml
   when: packer_builder_type is search('qemu')
+
+# Some OS might disable cloud-final service on boot (rhel 7).
+# Enable all cloud-init services on boot.
+# This sometimes reports service state as `UNKNOWN` and fails.
+# Retry 5 times to bypass above failure.
+- name: Make sure all cloud init services are enabled and running
```
Comment (detiber) [inline images/capi/ansible/roles/providers/tasks/main.yml:35]: I'm not sure that this needs to verify the services are running, I think just ensuring they are enabled should be sufficient here.

**Comment by EleanorRigby** [comment]
Sounds reasonable. I can change this.

File: images/capi/ansible/roles/providers/tasks/main.yml (hunk 1 above)
Comment (EleanorRigby) [reply images/capi/ansible/roles/providers/tasks/main.yml]: fixed.

File: images/capi/ansible/roles/providers/tasks/main.yml [hunk 2]
```diff
@@ -56,3 +56,20 @@
...
+- name: Make sure all cloud init services are enabled
+  service:
+    name: "{{ item }}"
+    enabled: yes
+  with_items:
+    - cloud-final
+    - cloud-config
+    - cloud-init
+    - cloud-init-local
+  register: cloud_init_enabled
```
Comment (codenrhoden) [inline images/capi/ansible/roles/providers/tasks/main.yml:73]: Just one question before giving this an `/approve`. Now that this is just enabled and not trying to start/restart, is it really the case that the state comes back `UNKNOWN` and the retries are needed? I'm thinking that just making sure the service is enabled won't fail, but could be wrong.

Comment (EleanorRigby) [reply images/capi/ansible/roles/providers/tasks/main.yml:73]: I saw this problem happening consistently on `RHEL` but never removed retries for this PR.  I can retry a build locally and remove if it works twice consecutively for some `AMI` .

You can add it later if you face some problems on RHEL

Comment (EleanorRigby) [reply images/capi/ansible/roles/providers/tasks/main.yml]: It works. Removed retries

**Pull request merged**